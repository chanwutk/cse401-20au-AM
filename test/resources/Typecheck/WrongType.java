class WrongType {
	public static void main(String[] args) {
		while (1)
			if (new C3())
				System.out.println(true + 1);
			else
				System.out.println(false && 2);
	}
}

class C1 {
	C1 c1;
	public C1 f(C2 x) {
		c1 = x;
		return x;
	}
}

class C2 extends C1 {
	C2 c2;
	public C2 f(C1 x) {
		c1 = x;
		c2 = x;
		c1 = this;
		return x.f(this);
	}
}

class C3 {
	public int f(int x, int y, int z) {
		int a;
		a = this.f(x, y, z);
		a = this.f(x, y);
		a = this.f(x, y, z, true);
		a = this.f(x, y, false);
		return a;
	}
}
