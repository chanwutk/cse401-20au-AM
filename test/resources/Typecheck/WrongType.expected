WrongType.java:3: error: incompatible types: int cannot be converted to boolean
WrongType.java:4: error: incompatible types: C3 cannot be converted to boolean
WrongType.java:5: error: bad operand types for binary operator '+'
  first type:  boolean
  second type: int
WrongType.java:7: error: bad operand types for binary operator '&&'
  first type:  boolean
  second type: int
WrongType.java:7: error: incompatible types: boolean cannot be converted to int
WrongType.java:33: error: method f in class C3 cannot be applied to given types;
  required: int,int,int
  found: int,int
  reason: actual and formal argument lists differ in length
WrongType.java:34: error: method f in class C3 cannot be applied to given types;
  required: int,int,int
  found: int,int,int,boolean
  reason: actual and formal argument lists differ in length
WrongType.java:35: error: incompatible types: boolean cannot be converted to int
WrongType.java:23: error: incompatible types: C1 cannot be converted to C2
WrongType.java:25: error: incompatible types: C1 cannot be converted to C2
